lib/loader/serialize/serial_module.cpp

```diff
@@ -25,9 +25,9 @@ Serializer::serializeModule(const AST::Module &Mod) const noexcept {
                    const AST::GlobalSection *, const AST::ExportSection *,
                    const AST::StartSection *, const AST::ElementSection *,
                    const AST::CodeSection *, const AST::DataSection *,
-                   const AST::DataCountSection *>;
+                   const AST::DataCountSection *, const AST::TagSection *>;
   std::vector<SecVariant> Sections;
-  Sections.reserve(Mod.getCustomSections().size() + 12);
+  Sections.reserve(Mod.getCustomSections().size() + 13);
```

Comment:
You can try variant_size https://en.cppreference.com/w/cpp/utility/variant/variant_size to remove this hard coded number.